redfish-core/lib/log_services.hpp

```diff
@@ -3407,4 +3470,965 @@ inline void requestRoutesPostCodesEntry(App& app)
+ *        xyz.openbmc_project.Association.Definitions::Associations of the
+ *        HardwareIsolation dbus entry object.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ *
+ * @return The redfish response with appropriate redfish properties of the
+ *         isolated hardware details into LogEntry schema if success else
+ *         nothing in redfish response.
+ */
+inline void fillIsolatedHwDetailsByObjPath(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const sdbusplus::message::object_path& dbusObjPath,
+    const size_t entryJsonIdx)
+{
+    // Fill Redfish uri of isolated hardware into "OriginOfCondition"
+    if (dbusObjPath.filename().find("unit") != std::string::npos)
+    {
+        // If Isolated Hardware object name contain "unit" then that unit
+        // is not modelled in inventory and redfish so the "OriginOfCondition"
+        // should filled with it's parent (aka FRU of unit) path.
+        getRedfishUriByDbusObjPath(asyncResp, dbusObjPath.parent_path(),
+                                   entryJsonIdx);
+    }
+    else
+    {
+        getRedfishUriByDbusObjPath(asyncResp, dbusObjPath, entryJsonIdx);
+    }
+
+    // Fill PrettyName of isolated hardware into "Message"
+    getPrettyNameByDbusObjPath(asyncResp, dbusObjPath, entryJsonIdx);
+}
+
+/**
+ * @brief API used to fill isolated hardware details into LogEntry schema
+ *        by using the given isolated dbus object.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details. If passing less than or equal 0 then,
+ *                            it will assume the given asyncResp jsonValue as
+ *                            a single entry json object. If passing greater
+ *                            than 0 then, it will assume the given asyncResp
+ *                            jsonValue contains "Members" to fill in the
+ *                            appropriate entry json object.
+ * @param[in] dbusObjIt - The DBus object which contains isolated hardware
+                         details.
+ *
+ * @return The redfish response with appropriate redfish properties of the
+ *         isolated hardware details into LogEntry schema if success else
+ *         failure response.
+ */
+inline void fillSystemHardwareIsolationLogEntry(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const size_t entryJsonIdx, GetManagedObjectsType::const_iterator& dbusObjIt)
+{
+    nlohmann::json& entryJson =
+        (entryJsonIdx > 0
+             ? asyncResp->res.jsonValue["Members"][entryJsonIdx - 1]
+             : asyncResp->res.jsonValue);
+
+    bool entryIsPresent{false};
```

Comment:
nitpick: I know this is the same but I have never seen this in upstream bmcweb 
All I see is 
https://github.com/openbmc/bmcweb/blob/97128e926c8253f02cf25d2ca9881cadc426df2c/include/sessions.hpp#L49

Reply:
Done - changed as requested.